pkg/wharfapi/requestHandler.go

```diff
@@ -126,7 +134,12 @@ func doRequest(from string, method string, URLStr string, body []byte, authHeade
 		return nil, fmt.Errorf("unexpected status code returned: %s", response.Status)
 	}
 
-	return &response.Body, nil
+	bytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return bytes, nil
```

Comment:
I think it simplified some things earlier, but I'll try reverting it back to `io.ReadCloser` 👍🏻

Reply:
@jilleJr
Changed back to using io.ReadCloser and json.NewDecoder.Decode.

Since it is getting read into an in-memory object anyways, why is the buffering beneficial here?

Can't quite wrap my head around it.
Is it because it would load all of it into memory first, and then have to allocate the same amount for the destination object at the same time to copy it over, therefore using 2 times the actual memory it takes?